### PR TITLE
Desktop Ledger app update

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

Update `hw-app-iota` for Ledger Blue compatibility 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [x] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: src/desktop/src/libs/SeedStore/Ledger.js [hunk 1]
```diff
@@ -138,7 +140,7 @@ class Ledger extends SeedStoreCore {
                 awaitTransaction: { address: options.address, value: transfers.reduce((a, b) => a + b.value, 0) },
             });
 
-            const trytes = await seed.signTransaction(transfers, options.inputs, remainder);
+            const trytes = await seed.perpareTransfers(transfers, options.inputs, remainder);
```
Comment (rajivshah3) [inline src/desktop/src/libs/SeedStore/Ledger.js]:
```suggestion
            const trytes = await seed.prepareTransfers(transfers, options.inputs, remainder);
```

File: src/shared/actions/transfers.js [hunk 2]
```diff
@@ -435,128 +437,138 @@ export const makeTransaction = (seedStore, receiveAddress, value, message, accou
...
-                    // Check if input count does not exceed maximum supported by the SeedStore type
-                    if (seedStore.maxInputs && inputs.inputs.length > seedStore.maxInputs) {
-                        throw new Error(Errors.MAX_INPUTS_EXCEEDED(inputs.inputs.length, seedStore.maxInputs));
-                    }
+                        //const maxInputs = await seedStore.getMaxInputs();
 
-                    transferInputs = get(inputs, 'inputs');
+                        // Check if input count does not exceed maximum supported by the SeedStore type
+                        if (maxInputs && inputs.inputs.length > maxInputs) {
+                            throw new Error(Errors.MAX_INPUTS_EXCEEDED(inputs.inputs.length, seedStore.maxInputs));
```
Comment (rajivshah3) [inline src/shared/actions/transfers.js]: Should this be `maxInputs` instead of `seedStore.maxInputs`?

File: src/shared/actions/transfers.js [hunk 3]
```diff
@@ -435,128 +437,138 @@ export const makeTransaction = (seedStore, receiveAddress, value, message, accou
...
-                    const valueTransfers = filter(
-                        map(accountState.transfers, (tx) => tx),
-                        (tx) => tx.transferValue !== 0,
-                    );
+        return Promise.resolve(isLastTritZero(address))
+            .then((lastTritIsZero) => {
+                if (!lastTritIsZero) {
+                    throw new Error(Errors.INVALID_LAST_TRIT);
+                }
+                return typeof seedStore.getMaxInputs === 'function' ? seedStore.getMaxInputs() : Promise.resolve(0);
```
Comment (rajivshah3) [inline src/shared/actions/transfers.js:445]: Won't `seedStore.getMaxInputs` always be a function?

File: src/shared/actions/transfers.js [hunk 4]
```diff
@@ -435,128 +437,138 @@ export const makeTransaction = (seedStore, receiveAddress, value, message, accou
...
-                    }
+                        if (isSendingToAnyInputAddress) {
+                            throw new Error(Errors.CANNOT_SEND_TO_OWN_ADDRESS);
+                        }
 
-                    // Check if input count does not exceed maximum supported by the SeedStore type
-                    if (seedStore.maxInputs && inputs.inputs.length > seedStore.maxInputs) {
-                        throw new Error(Errors.MAX_INPUTS_EXCEEDED(inputs.inputs.length, seedStore.maxInputs));
-                    }
+                        //const maxInputs = await seedStore.getMaxInputs();
```
Comment (cvarley100) [inline src/shared/actions/transfers.js]: Comment can be deleted

**Review by cvarley100 — APPROVED** [review]

**Pull request merged**